src/output/plugins/httpd/HttpdClient.hxx

```diff
@@ -111,6 +117,11 @@ class HttpdClient final
 	 */
 	unsigned metadata_fill = 0;
 
+	/**
+	 * Provided password (using HTTP basic auth)
+	 */
+	std::string *provided_password;
```

Comment:
A pointer to a `std::string`? That makes my head hurt.
One way to make the code simpler and safer would be assuming an empty string means no password was provided; but if you really want to know the difference between "no password" and "empty password", then `std::optional<std::string>` would be better. That would be safer *and* simpler. Though I think it would be somewhat overengineered. Empty string is fine for me.

Reply:
Okay, empty password in config file now behaves exactly as no password.